save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPlugin.kt

```diff
@@ -123,125 +97,101 @@ class WarnPlugin(
-                    " and there are ${extraFlagsList.size} different sets of flags inside it, namely $extraFlagsList"
-        }
-        val extraFlags = extraFlagsList.singleOrNull() ?: ExtraFlags("", "")
+        if (expectedWarningsMap.isEmpty()) warnMissingExpectedWarnings(warnPluginConfig, generalConfig, originalPaths)
 
-        if (expectedWarningsMap.isEmpty() && warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.IN_PLACE) {
-            logWarn(
-                "No expected warnings were found using the following regex pattern:" +
-                        " [${generalConfig.expectedWarningsPattern}] in the test files: $paths." +
-                        " If you have expected any warnings - please check 'expectedWarningsPattern' or capture groups" +
-                        " in your 'save.toml' configuration"
-            )
-        } else if (expectedWarningsMap.isEmpty() && warnPluginConfig.expectedWarningsFormat == ExpectedWarningsFormat.SARIF) {
-            logWarn(
-                "No expected warnings were found when inspecting files ${warnPluginConfig.expectedWarningsFileName}" +
-                        " for test files: $paths." +
-                        " If you have expected any warnings - please make sure SARIF files exist, have correct name and contain" +
-                        " relevant warnings."
-            )
-        }
+        val cmdExecutor = CmdExecutorWarn(
+            generalConfig,
+            copyPaths,
+            extraFlagsExtractor,
+            pb,
+            warnPluginConfig,
+            testConfig,
+            fs,
+        )
 
-        // joining test files to string with a batchSeparator if the tested tool supports processing of file batches
-        // NOTE: SAVE will pass relative paths of Tests (calculated from testRootConfig dir) into the executed tool
-        val fileNamesForExecCmd =
-                warnPluginConfig.wildCardInDirectoryMode?.let {
-                    // a hack to put only the root directory path to the execution command
-                    // only in case a directory mode is enabled
-                    var testRootPath = copyPaths[0].parent ?: ".".toPath()
-                    while (testRootPath.parent != null && testRootPath.parent!!.name != tmpDirName) {
-                        testRootPath = testRootPath.parent!!
-                    }
-                    "$testRootPath$it"
-                } ?: copyPaths.joinToString(separator = warnPluginConfig.batchSeparator!!)
+        val execCmd = cmdExecutor.constructExecCmd(tmpDirName)
 
-        logDebug("Constructed file name for execution for warn plugin: $fileNamesForExecCmd")
-        val execFlagsAdjusted = resolvePlaceholdersFrom(warnPluginConfig.execFlags, extraFlags, fileNamesForExecCmd)
-        val execCmd = "${generalConfig.execCmd} $execFlagsAdjusted"
-        val time = generalConfig.timeOutMillis!!.times(copyPaths.size)
+        try {
```

Comment:
Why do we need `try` on this whole block, while catching only `pb` exceptions? Could it be done only for
`cmdExecutor.executeCommandAndGetTestResults` like before?

Reply:
unfortunately not, yeah, I tried to refactor this in different ways. 
But we need to set `execCmd` in case of exception in catch block, so it cannot be handled inside :(

This is the best variant unfortunately

Reply:
fixed